d3b-center/image-deid-etl
# Update hardcoded s3 bucket & CSV file name to source from environment and user input

## Overview

The main changes are that the s3 bucket path is now an environment variable, and the CSV table required for subject mapping information is an optional user input. The main assumption/constraint is that the CSV file will always live at the s3 bucket path.

Also removes `boto3` dependency (reads CSV file directly from s3 via `pandas`), and some unnecessary/old code related to logging that is not being used.

Resolves #18 

### Checklist

- [x] s3 bucket path is defined as environment variable
- [x] `s3_path` and `s3_path_to_csv` are constructed based on environment variables and user inputs (program, site)
- [x] path to CSV is input to pandas read_csv
- [x] updated README accordingly to include S3_BUCKET definition

### Notes

An assumption/constraint is that the CSV file will always live at the s3 bucket path.

## Testing Instructions

* update your .env file or run:
`export S3_BUCKET="s3://d3b-phi-data-prd/imaging/"`
* if necessary, update local db
`./image_deid_etl check --mark-processed`
```
DELETE
FROM processed_uuids
WHERE UUID IN ('f90cd746-10017fbb-44472a28-391f1a52-3186cf1a',
               'd1b90c78-d9b579f2-a4664679-7ede472b-978d73a2');
```
* run test script
`./scripts/test`
